juicebox_scripts/juicebox_assembly_purger.py

```diff
@@ -73,30 +73,36 @@ def filter_assembly(exclude, input_assembly, output_assembly, **kwargs):
         purged_names = set([])
         purged_indices = set([])
         purged_from_scaffolds = set([])
+        index_map = {}
         for line in input:
             if line.startswith(">"):
                 name, num, length = line.strip().split()
                 name = name.replace(">", "")
                 if name in exclude:
                     purged_names.add(name)
                     purged_indices.add(num)
+                    index_map[num] = None
                     continue
                 else:
                     outstring = ">{} {} {}\n".format(name, index, length)
                     output.write(outstring)
+                    index_map[num] = str(index)
                     index += 1
             else:
                 outlist = []
                 contigs = line.strip().split()
-                for contig_num in contigs:
-                    if contig_num.replace("-", "") in purged_indices:
-                        purged_from_scaffolds.add(contig_num.replace("-", ""))
+                for orient_num in contigs:
+                    orientation = "-" if orient_num.startswith("-") else ""
+                    num = orient_num.replace("-", "")
+                    if index_map[num] is None:
```

Comment:
would it make sense to have a more explicit test here for whether the index is in purged?

Reply:
Yeah, that change can help avoid a NameError. Fixed.